blacken_docs.py

```diff
@@ -39,6 +39,24 @@
     r'(?P<code>(^((?P=indent) +.*)?(\n|$))+)',
     re.MULTILINE,
 )
+RST_DOCTEST_RE = re.compile(
+    r'(?P<before>'
+    r'^(?P<indent> *)\.\. (testsetup|testcleanup|testcode)::.*\n'
+    r'((?P=indent) +:.*\n)*'
+    r'\n*'
+    r')'
+    r'(?P<code>(^((?P=indent) +.*)?\n)+)',
+    re.MULTILINE,
+)
+RST_DOCTEST_PYCON_RE = re.compile(
+    r'(?P<before>'
+    r'(?P<indent> *)\.\. (doctest)::.*\n'
+    r'((?P=indent) +:.*\n)*'
+    r'\n*'
+    r')'
+    r'(?P<code>(^((?P=indent) +.*)?(\n|$))+)',
+    re.MULTILINE,
+)
```

Comment:
these should be combined with the existing `rst` /  `pycon` regexes since they represent the same operations

Reply:
I didn't want to make the current ones uglier since the `doctest` ones can have any directive argument, but I'll combine it.  Let me know if you have a better alternative than just doing an "or" as I've done in the new code.